### ISSUE TITLE
TFS 2017 - DeployTestAgent failed - caused infinite reboot loop

### ISSUE THREAD

**Issue by ISAF87** [Microsoft/vsts-tasks#5129]
25T22:03:39.3077144Z ##[error]Error occured on '<machinename>:5985'. Details : 'Exception calling "StopService" : "Not found "'. For troubleshooting, refer https://aka.ms/remotevstest.
2017-08-25T22:03:39.3077144Z ##[error]System.Exception: Error occured on '<machinename>:5985'. Details : 'Exception calling "StopService" : "Not found "'. For troubleshooting, refer https://aka.ms/remotevstest.
2017-08-25T22:03:39.3077144Z ##[warning]For troubleshooting, refer https://aka.ms/remotevstest.
2017-08-25T22:03:39.3077144Z ##[error]System.InvalidOperationException: Deploy Test Agent task has failed on all machines. Check the logs for more details.
2017-08-25T22:03:39.3077144Z    at Microsoft.TeamFoundation.DistributedTask.Task.DistributedTestAutomation.DeployTestAgent.Run()
2017-08-25T22:03:39.3077144Z    at Microsoft.TeamFoundation.DistributedTask.Task.DistributedTestAutomation.TaskRunner.Run(Dictionary`2 arguments, VssConnection connection, TaskType taskType)
2017-08-25T22:03:39.3077144Z    at Microsoft.TeamFoundation.DistributedTask.Task.DistributedTestAutomation.InvokeDeployTestAgentCmdlet.ProcessRecord()
2017-08-25T22:03:39.3077144Z ##[error]PowerShell script completed with 2 errors. 

It should deploy as interactive process, without updating the agent. I do expect a reboot because of the session/credentials. But after this it kept rebooting the client, even after the actual release was stopped. Shouldn't a autologoncount parameter be set in the register to prevent this from happening? Can we do any custom steps to prevent this behaviour?

**Comment by nigurr** [comment]
Can you please let us know the version of Deploy Test Agent task? and please share the build logs.

**Comment by ISAF87** [comment]
Please see the attached file for logs.
The task version number is 1.*

[deploy test agent build logs.txt](https://github.com/Microsoft/vsts-tasks/files/1256962/deploy.test.agent.build.logs.txt)

**Comment by nigurr** [comment]
@ISAF87 This has been addressed in TFS 2017 Update 2 if you are using On-Prem.
And same is available with v2 version of task in VSTS

Please try out the same and let us know if you are still facing the issue. 
For more info - please check this blog https://blogs.msdn.microsoft.com/devops/2017/07/03/deploy-visual-studio-test-agent-task-test-machine-rebooting-scenarios/

**Comment by ISAF87** [comment]
Can you briefly explain the cause, so I can clarify on the project what is happening and why we should upgrade?

**Comment by nigurr** [comment]
@ISAF87 For running automated UI tests, we reboot the machine to get interactive session. Due to corporate group policies/Registry settings, we may not be able to auto logon to that machine for running the tests. 
In this case, we will end up rebooting continuously. In later releases, we have added proper warnings and reboot limits to let users inform why we failed to get "auto logon" session for running the tests.
This is required to run the tests.
